tests/run_tests.py

```diff
@@ -0,0 +1,48 @@
+from unittest import TestSuite, TestLoader, TestCase
+import unittest
+import sys
+
+
+keys = sys.argv[1:]
+
+loader = TestLoader()
+all_tests = loader.discover('coconut.tests')
+
+
+def key_in_test(key, test):
+    test_suite = TestSuite()
+    if isinstance(test, TestSuite):
+        if test.countTestCases() == 0:
+            return []
+        for test_case in test:
+            test_suite.addTests(key_in_test(key, test_case))
+        return test_suite
+    elif isinstance(test, TestCase):
+        if key.lower() in str(test).lower():
+            test_suite.addTest(test)
+    else:
+        raise TypeError('Variable "test" is not an instance of TestSuite or TestCase')
```

Comment:
Do you see a reason why an element of all_tests, obtained using `loader.discover` would not be an instance of TestSuite or TestCase?

Reply:
No it should not happen. When implementing there was a change of running into an empty list, but now the check can be removed.